orchestrator-salt/src/main/resources/salt/salt/metering/scripts/metering_package_manager.sh

```diff
@@ -0,0 +1,316 @@
+#!/bin/sh
+
+METERING_PACKAGE_NAME="thunderhead-metering-heartbeat-application"
+METERING_FILE_NAME="$METERING_PACKAGE_NAME-0.1-SNAPSHOT.x86_64.rpm"
+: ${METERING_SERVICE_DELIVERY_URL:="https://cloudera-service-delivery-cache.s3.amazonaws.com/$METERING_PACKAGE_NAME/clients/$METERING_FILE_NAME"}
+# todo: provide available archive.cloudera.com rpm release location
+: ${METERING_ARCHIVE_URL:="https://cloudera-service-delivery-cache.s3.amazonaws.com/$METERING_PACKAGE_NAME/clients/$METERING_FILE_NAME"}
+: ${LOGFILE_FOLDER:="/var/log/metering-upgrade"}
+MINION_SCRIPTS_FOLDER="/opt/salt/scripts"
+CURL_CONNECT_TIMEOUT=60
+SALT_CMD_TIMEOUT=120
+MIN_BYTES=1000000
+
+readlinkf(){
+  perl -MCwd -e 'print Cwd::abs_path shift' "$1";
+}
+
+if [ "$(uname -s)" = 'Linux' ]; then
+  SCRIPT_LOCATION=$(readlink -f "$0")
+else
+  SCRIPT_LOCATION=$(readlinkf "$0")
+fi
+
+function print_help() {
+  cat << EOF
+   Usage: [<command>] [<arguments with flags>]
+   commands:
+     upgrade              download and upgrade metering heartbeat agent
+     upgrade-only         upgrade metering heartbeat agent without download
+     download             download (only) metering heartbeat agent from salt master
+     help                 print usage
+   upgrade command arguments:
+     -d, --from-date      <yyyy-mm-dd>       Install RPM if the installation date is lower than this value.
+     -f, --rpm-file       <LOCAL RPM FILE>   Local RPM file to be installed (required for upgrade only)
+     -u, --custom-rpm-url <CUSTOM RPM URL>   Custom URL that will be used instead of the default archive URL
+EOF
+}
+
+function do_exit() {
+  local code=$1
+  local message=$2
+  if [[ "$message" == "" ]]; then
+    info "Exit code: $code"
+  else
+    info "Exit code: $code, --- STATUS MESSAGE --- $message --- STATUS MESSAGE ---"
+  fi
+  exit $code
+}
+
+function init_logfile() {
+  mkdir -p $LOGFILE_FOLDER
+  local timestamp=$(date +"%Y%m%d-%H%M%S")
+  LOGFILE="$LOGFILE_FOLDER/metering-upgrade-${timestamp}.log"
+  touch $LOGFILE
+  cleanup_old_logs
+  info "The following log file will be used: $LOGFILE"
+}
+
+function init_salt_prefix() {
```

Comment:
if you really need salt to get the proxy stuff you might depend on `/usr/bin/activate_salt_env` to get the correct path

Reply:
i would like to avoid to context change especially as some env vars are in use